### PR TITLE
feat: allow users to hide Safe Pass promotion

### PR DESCRIPTION
**Pull request by schmanu**
## How this PR fixes it
Adds "Don't show again" button to Safe Pass promotion.

## How to test it
- Go to the dashboard on a mainnet Safe
- Hide the promotion

## Screenshots
Screenshot 2024-07-22 at 15 19 46
Screenshot 2024-07-22 at 15 19 39

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/dashboard/ActivityRewardsSection/index.tsx [hunk 1]
```diff
@@ -39,10 +40,12 @@ const ActivityRewardsSection = () => {
   const isDarkMode = useDarkMode()
   const router = useRouter()
 
+  const [widgetHidden = false, setWidgetHidden] = useLocalStorage<boolean>('show-activity-rewards-section')
```
Comment (katspaugh) [inline src/components/dashboard/ActivityRewardsSection/index.tsx:43]: Please use camelCase for LS flags and extract the name into a variable.

File: src/components/dashboard/ActivityRewardsSection/index.tsx [hunk 2]
```diff
@@ -101,6 +96,26 @@ const ActivityRewardsSection = () => {
...
+              <NextLink
+                href="https://safe.global/pass"
+                target="_blank"
+                passHref
+                rel="noreferrer"
+                onClick={onClick}
+                style={{ marginLeft: 'auto' }}
+              >
+                <Button variant="text">Learn more</Button>
+              </NextLink>
```
Comment (katspaugh) [inline src/components/dashboard/ActivityRewardsSection/index.tsx:116]: It would be good to track these buttons/links.

Comment (katspaugh) [reply src/components/dashboard/ActivityRewardsSection/index.tsx:116]: Oh, and please separate them with new lines.

File: src/services/analytics/events/overview.ts [hunk 3]
```diff
@@ -145,6 +145,16 @@ export const OVERVIEW_EVENTS = {
     action: 'Open activity app from widget',
     category: OVERVIEW_CATEGORY,
   },
+  HIDE_ACTIVITY_APP_WIDGET: {
+    event: EventType.CLICK,
```
Comment (katspaugh) [inline src/services/analytics/events/overview.ts:149]: This the default, not necessary to pass.

File: src/services/analytics/events/overview.ts [hunk 4]
```diff
@@ -145,6 +145,16 @@ export const OVERVIEW_EVENTS = {
...
+    action: 'Hide activity app widget',
+    category: OVERVIEW_CATEGORY,
+  },
+  OPEN_LEARN_MORE_ACTIVITY_APP: {
+    event: EventType.CLICK,
+    action: 'Open learn more page from widget',
```
Comment (katspaugh) [inline src/services/analytics/events/overview.ts:155]:
```suggestion
    action: 'Activity app learn more'
```

**Review by katspaugh — APPROVED** [review]

**Comment by francovenica** [comment]
@schmanu It seems the PR didn't actually deployed. The environment doesn't work, throws a 404. Maybe it has to do with what it was discussed [here](https://5afe.slack.com/archives/C03DAGWJCR5/p1721656483196489) ?

**Comment by schmanu** [comment]
@francovenica I merged dev into this and it works again. Can you re-test?

**Comment by francovenica** [comment]
Thanks for re-building.

The button works fine, but it looks kinda out of place for the mobile viewport for being next to the "load more". 
image

I'd say that it should look like this, where if you reduce the view port even more the "load more" goes down a bit. but the button probably should be centered if this is the case:
image

@TanyaEfremova Any suggestion?

**Comment by TanyaEfremova** [comment]
Fair comment! Is it possible to implement the following rules to the mobile button groups in widgets:
- Display them 1 below another and stretch in full width
- _Learn more_ can belong to the stepper component and be shown below

[See example ](https://www.figma.com/design/ptTs6lDBeUuLNySroJ5PiF/Web-Master-File?node-id=1010-42756&t=iY5MRrf21Sjfm4QA-1)

cc @schmanu

**Comment by schmanu** [comment]
@francovenica I adjusted it according to Tanyas comments. Can you re-check?

**Comment by francovenica** [comment]
Looks nice!

image

**Pull request merged**